src/components/SettingsDialog.tsx

```diff
@@ -0,0 +1,155 @@
+import { useEffect, useMemo, useState } from 'react';
+import { X } from 'lucide-react';
+import { Button } from './ui/Button';
+import { utils } from '@typeberry/state-merkleization';
+
+interface SettingsDialogProps {
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+const SettingsDialog = ({ isOpen, onClose }: SettingsDialogProps) => {
+  const [selectedGpVersion, setSelectedGpVersion] = useState<string>('');
+  const [selectedSuite, setSelectedSuite] = useState<string>('');
+  const [applyError, setApplyError] = useState<string | null>(null);
+  const [applySuiteError, setApplySuiteError] = useState<string | null>(null);
+
+  const gpOptions = useMemo(() => {
+    const src = utils?.GpVersion as Record<string, unknown> | undefined;
+    if (!src) return [] as string[];
+    const values = Object.values(src).filter((v) => typeof v === 'string') as string[];
+    return [...new Set(values)];
+  }, []);
+
+  const suiteOptions = useMemo(() => {
+    const src = utils?.TestSuite as Record<string, unknown> | undefined;
+    if (!src) return [] as string[];
+    const values = Object.values(src).filter((v) => typeof v === 'string') as string[];
+    return [...new Set(values)];
+  }, []);
+
+  const hasOverride = typeof utils?.Compatibility?.override === 'function';
+  const hasOverrideSuite = typeof utils?.Compatibility?.overrideSuite === 'function';
+
+  useEffect(() => {
+    if (!isOpen) return;
+    setApplyError(null);
+    setApplySuiteError(null);
+
+    setSelectedGpVersion(gpOptions[0] ?? '');
+    setSelectedSuite(suiteOptions[0] ?? '');
+  }, [isOpen, gpOptions, suiteOptions]);
+
+  if (!isOpen) return null;
+
+  const applyGpVersion = (value: string) => {
+    setSelectedGpVersion(value);
+    setApplyError(null);
+    const fn = utils?.Compatibility?.override;
+    if (typeof fn === 'function') {
+      try {
+        fn(value as unknown as never);
+        window.location.reload();
```

Comment:
Window shouldn't be reloaded, we just need to make sure that the top-level component is reloaded.